### PR TITLE
Improve elixir xdr modules

### PR DESCRIPTION
**Pull request by marioyustikommit**
## Improvements for all XDR types

- Documentation improvements.
- Improvements in the implementation of some xdr types.
- Some improvements in the unit tests
- Improvements in the definition of exceptions into `XDR.Error` module.

### CODE REVIEW

File: lib/error/error.ex [hunk 1]
```diff
@@ -1,729 +1,590 @@
...
+      "The value which you try to encode is not an integer"
+      |> new()
     end
 
     def exception(:not_binary) do
-      msg =
-        "The value which you try to decode must be a binary value, for example: <<0, 0, 0, 2>>"
-
-      %XDR.Error.UInt{message: msg}
+      "The value which you try to decode must be a binary value, for example: <<0, 0, 0, 2>>"
```
Comment (juanhurtado10) [inline lib/error/error.ex:58]: @marioyustikommit As a tip when your pipe operator has just one step the recommendation is to write the function like:  
````
function(param)
````

File: README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Process XDR types based on the [RFC4506](https://www.ietf.org/rfc/rfc4506.txt).
 ```elixir
 def deps do
   [
-    {:elixir_xdr, "~> 0.1.0"}
+    {:elixir_xdr, "~> 0.1.3"}
```
Comment (juanhurtado10) [inline README.md:16]: @marioyustikommit 0.1.4 is the next version. Could you please also update the readme file (only if it is required)

File: lib/xdr/bool.ex [hunk 3]
```diff
@@ -1,78 +1,78 @@
...
   @doc """
-  This function is in charge of decode XDR data which represents a boolean value
-
-  returns the boolean decoded from an XDR value
+  Decode the Boolean in XDR format to a `XDR.Bool` structure.
+  If the binary is not a valid boolean, an exception is raised.
   """
-  @spec decode_xdr!(bytes :: binary, struct :: t | any) :: {t(), binary}
-  def decode_xdr!(bytes, struct \\ %XDR.Bool{declarations: @boolean})
+  @spec decode_xdr!(bytes :: binary, struct :: t() | any) :: {t(), binary}
```
Comment (juanhurtado10) [inline lib/xdr/bool.ex:69]: @marioyustikommit what is the meaning of t() as a typespec

File: lib/error/error.ex (hunk 1 above)
Comment (marioyustikommit) [reply lib/error/error.ex]: Thank you Juan, it is done!

File: README.md (hunk 2 above)
Comment (marioyustikommit) [reply README.md]: Yes, I updated the README and the Guides of Void and Union.

File: lib/xdr/bool.ex (hunk 3 above)
Comment (marioyustikommit) [reply lib/xdr/bool.ex]: Sorry, I was investigating and I realized that It is a bad practice that I adopted. I fixed it.

**Comment by juanhurtado10** [comment]
@marioyustikommit just one thing left. We need to update the Union section in the Readme its not very clear for people even for me. Could you please? Thank you!

**Pull request merged**